iOS8-HW12-Yuliya Khrebtova/Sources/ViewController.swift

```diff
@@ -0,0 +1,230 @@
+        let timeLabel = UILabel()
+        timeLabel.text = workTime.minuteSecondMS
+        timeLabel.textAlignment = .center
+        timeLabel.textColor = .systemMint
+        timeLabel.font = .systemFont(ofSize: 40, weight: .regular)
+        timeLabel.translatesAutoresizingMaskIntoConstraints = false
+        return timeLabel
+    }()
+
+    private lazy var progressBarView: UIView = {
+        let progressBarView = UIView()
+        progressBarView.translatesAutoresizingMaskIntoConstraints = false
+        return progressBarView
+    }()
+
+    // MARK: - Lifecycle
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        setBackgroundColor()
+        setupHierarchy()
+        setupLayout()
+
+        count = workTime
+    }
+
+    // MARK: - Setup
+    private func setBackgroundColor() {
+        view.backgroundColor = .black
+    }
+
+    private func setupHierarchy() {
+        view.addSubview(label)
+        view.addSubview(progressBarView)
+        view.addSubview(button)
+        view.addSubview(timeLabel)
+        backgroundCircular()
+        animationCircular(startGradientColor: .systemMint, endGradientColor: .cyan)
+    }
+
+    private func setupLayout() {
+        NSLayoutConstraint.activate([
+            label.centerXAnchor.constraint(equalTo: view.centerXAnchor),
+            label.topAnchor.constraint(equalTo: view.topAnchor, constant: 100),
+
+            progressBarView.centerXAnchor.constraint(equalTo: view.centerXAnchor),
+            progressBarView.centerYAnchor.constraint(equalTo: view.centerYAnchor),
+            progressBarView.heightAnchor.constraint(equalToConstant: progressBarSize),
+            progressBarView.widthAnchor.constraint(equalToConstant: progressBarSize),
+
+            button.centerXAnchor.constraint(equalTo: view.centerXAnchor),
+            button.bottomAnchor.constraint(equalTo: view.bottomAnchor, constant: -100),
+            button.heightAnchor.constraint(equalToConstant: 50),
+            button.widthAnchor.constraint(equalToConstant: 50),
+
+            timeLabel.centerXAnchor.constraint(equalTo: view.centerXAnchor),
+            timeLabel.centerYAnchor.constraint(equalTo: view.centerYAnchor)
+        ])
+    }
+
+    // MARK: - Actions
+
+    @objc private func pressedButton() {
+        let playImage = UIImage(systemName: "play.fill", withConfiguration: configSymbol)
+        let pauseImage = UIImage(systemName: "pause.fill", withConfiguration: configSymbol)
+
+        isStartButtonPressed = !isStartButtonPressed
+        if isStartButtonPressed {
+            timer.invalidate()
+
+            if !isStarted {
+                progressBarAnimation()
+                isStarted = true
+            }
+
+            shapeLayer.resumeAnimation()
+            button.setImage(pauseImage, for: .normal)
+            timer = Timer.scheduledTimer(timeInterval: 0.001, target: self, selector: #selector(timerAction), userInfo: nil, repeats: true)
```

Comment:
Если получается слишком длинный вызов переносим параметры
```suggestion
            timer = Timer.scheduledTimer(timeInterval: 0.001, 
                                         target: self, 
                                         selector: #selector(timerAction), 
                                         userInfo: nil,
                                         repeats: true)
```